### PR TITLE
Update backend rest api starter kit

### PR DESCRIPTION
**Pull request by Chidi3b3r3**
- adds typescript support
- add prettier and run on circle ci builds
- removes eslint
- upgrades babel to version 7
- updates readme

# [Trello card](https://trello.com/c/k1BRhfuE/262-murcul-update-backend-rest-api-starter-kit)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,6 +36,11 @@ jobs:
             - node_modules
           key: v1-dependencies-{{ checksum "package.json" }}
 
+        # run prettier!
```
Comment (ziahamza) [inline .circleci/config.yml]: @Chidi3b3r3 

Mistake for the spaces.

File: README.md [hunk 2]
```diff
@@ -35,12 +33,10 @@ npm install
 yarn
 ```
 
-_If you don't use [Yarn](https://yarnpkg.com/) you can just replace `yarn` with `npm` in the commands that follow._
-
 Then you can begin development:
 
 ```sh
-yarn run dev
+npm run dev
```
Comment (ziahamza) [inline README.md]: hmm if we are going to nuke yarn, @Chidi3b3r3 can you then also nuke `yarn.lock` file?

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+
+{
```
Comment (ziahamza) [inline tsconfig.json]: @Chidi3b3r3 lets enable "script" option here as its much more stricter. 

Also @Chidi3b3r3 we want to convert existing files into typescript properly (they are not a lot, just general small files)

File: tsconfig.json [hunk 4]
```diff
@@ -1,22 +1,73 @@
...
   "compilerOptions": {
-    "module": "commonjs",
-    "esModuleInterop": true,
-    "target": "es6",
-    "noImplicitAny": true,
-    "moduleResolution": "node",
-    "sourceMap": true,
-    "outDir": "dist",
-    "baseUrl": ".",
+    /* Basic Options */
```
Comment (ziahamza) [inline tsconfig.json]: @Chidi3b3r3 I dont think we need to keep the commented items. Just add the items that matter and remove the rest.

File: tsconfig.json [hunk 5]
```diff
@@ -1,22 +1,73 @@
...
+    /* Module Resolution Options */
+    // "moduleResolution": "node",            /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
+    "baseUrl": ".",                       /* Base directory to resolve non-absolute module names. */
     "paths": {
-      "*": [
-        "node_modules/*",
-        "src/typings/*"
-      ]
-    }
+        "*": [
```
Comment (ziahamza) [inline tsconfig.json]: Indentation issue? the indentation in the file is un-even

File: README.md (hunk 2 above)
Comment (Chidi3b3r3) [reply README.md]: Done.

File: .circleci/config.yml [hunk 6]
```diff
@@ -29,20 +29,25 @@ jobs:
...
 
       - save_cache:
           paths:
             - node_modules
           key: v1-dependencies-{{ checksum "package.json" }}
 
+        # run prettier!
+      - run:
+          name: Prettier
+          command: npm run prettier
```
Comment (ziahamza) [inline .circleci/config.yml]: @Chidi3b3r3 So this command will never fail. We want the CI to fail if a file is not properly prettified. The actual command used would be `--list-different`. See: https://prettier.io/docs/en/cli.html

So something like: `npm run prettier -l`

Comment (ziahamza) [reply .circleci/config.yml]: aaah nvm, you already do that in package json @Chidi3b3r3 :D 👍

Comment (Chidi3b3r3) [reply .circleci/config.yml]: That is what `npm run prettier` would run as `prettier` has been defined as an npm script `prettier --list-different \"{src,test}/**/*{js,ts}\"`. I can move it here though. I just wanted here to be clean reason I left it as an npm script.

Comment (ziahamza) [reply .circleci/config.yml]: Totally agree, your solution is cleaner @Chidi3b3r3 .

The command name can be more descriptive, like `npm run prettier-check` ;) rather then `npm run prettier`

**Pull request merged**